### ISSUE TITLE
Add `Expect.isMatch actualString regexString`

### ISSUE THREAD

**Issue by haf** [haf/expecto#132]
And another `Expect.isRegexMatch : string -> Regex -> unit`

Aim: let the user expect a string to match a regex.

Also add to `Expect.Flip`.

Maybe this?

```fsharp
module Expect =
  let isMatch actual rstring message =
    match Regex.``match`` rstring actual with
    | None ->
      Tests.failtestf "%s. Expected %s to match /%s/" message actual rstring
    | Some x ->
      x
```

Plus the rest + docs?

**Comment by MNie** [comment]
@haf isNotRegexMatch also?

**Comment by haf** [comment]
Yes, that makes sense. And perhaps one that takes a callback that takes the matched groups.

**Comment by MNie** [comment]
@haf 
>And perhaps one that takes a callback that takes the matched groups.`

You mean something like this?

    let isRegexMatchGroups actual pattern doSomethingWithMatch message =
        let groups = Regex.Match(actual, pattern).Groups
        match doSomethingWithMatch groups with
        | Some x -> ()
        | None -> Tests.failtestf "%s. Expected %s to match pattern: /%s/" message actual pattern

**Comment by haf** [comment]
Thinking about it, perhaps the best version is to return the groups if there's a match. Let's merge what you've already done, it's good enough for now, and we can ponder the groups ;)

**Comment by haf** [comment]
Fixed with #133
